TestProxyHandler fails with test.count greater than 1

This test asserts on the value of a global counter. When run with -test.count > 1, it passes once and fails all subsequent runs.

### Failure cluster [d12b09d3e6d011d68e49](https://go.k8s.io/triage#d12b09d3e6d011d68e49)

##### Error text:
```
=== RUN   TestProxyHandler
E1201 04:12:11.066537  112514 handler_proxy.go:148] error resolving test-ns/bad-service: unresolveable
2021/12/01 04:12:11 http: TLS handshake error from 127.0.0.1:58306: remote error: tls: bad certificate
2021/12/01 04:12:11 http: TLS handshake error from 127.0.0.1:50404: remote error: tls: bad certificate
    handler_proxy_test.go:380: expected the x509_missing_san_total to be 1, but it's 6
--- FAIL: TestProxyHandler (0.10s)

```
#### Recent failures:
[12/15/2021, 4:50:06 AM ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1471054497823330304)
[12/14/2021, 10:50:07 PM ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1470963899351175168)
[12/14/2021, 4:49:15 PM ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1470873066585198592)
[12/14/2021, 10:48:48 AM ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1470782402564460544)
[12/14/2021, 4:49:33 AM ci-kubernetes-generate-make-test-count10](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-generate-make-test-count10/1470691804750811136)

/kind failing-test
/sig api-machinery

--- a/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy_test.go
+++ b/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy_test.go
@@ -301,6 +301,7 @@ func TestProxyHandler(t *testing.T) {
 	target := &targetHTTPHandler{}
 	for name, tc := range tests {
 		target.Reset()
+		legacyregistry.Reset()
 
 		func() {
 			targetServer := httptest.NewUnstartedServer(target)

